include/deal.II/hp/fe_collection.h

```diff
@@ -274,6 +274,45 @@ namespace hp
     unsigned int
     find_least_face_dominating_fe(const std::set<unsigned int> &fes) const;
 
+    /**
+     * Try to find a most face dominating finite element inside the subset of
+     * fe_indices @p fes as part of this FECollection. For example, if an
+     * FECollection consists of `{FE_Q(1),FE_Q(2),FE_Q(3),FE_Q(4)}` elements
+     * and we are looking for the most face dominating finite element among the
+     * last two elements of this collection (i.e., @p fes is `{2,3}`), then the
+     * answer is FE_Q(4) and therefore this function will return its index in
+     * the FECollection, namely `3`.
+     *
+     * This function differs from find_least_face_dominating_fe() in such a way
+     * that it only considers the subset @p fes, instead of finding a dominating
+     * element in the whole FECollection.
+     *
+     * For the purpose of this function by domination we consider either
+     * FiniteElementDomination::Domination::this_element_dominates or
+     * FiniteElementDomination::Domination::either_element_can_dominate;
+     * therefore the element can dominate itself. Thus, if an FECollection
+     * contains `{FE_Q(1),FE_Q(2),FE_Q(3),FE_Q(4)}` and @p fes only has
+     * a single element `{3}`, then the function returns 3.
+     *
+     * If the function is not able to find a finite element, the function
+     * returns numbers::invalid_unsigned_int .
+     *
+     * For an object, such as a line or a quad iterator, determine
+     * the fe_index of the most dominating finite element that
+     * lives on this object.
+     *
+     * Note that for the cases like when FECollection consists of `{FE_Nothing
+     * x FE_Nothing, FE_Q(1)xFE_Q(2), FE_Q(2)xFE_Q(1)}` with @p fes as `{1}`,
```

Comment:
i think you need to adjust `@p fes` in this discussion to fit this function. Otherwise, in the current state it should return `1` because you are looking for dominating FEs within `{1}`.